pkg/listwatch/listwatch.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2019 Istio Authors
```

Comment:
A couple of notes:
I am not sure that this is the right abstraction to create. The fact that you had to come up with your own demultiplexing scheme for the ListOptions is an indicator of that. IMHO coming up with an infrastructure that deals with the life-cycle aspects of ListerWatchers that talk to a common SharedInformerFactory is a better approach.

Also, I don't think we want this in top level pkg. Moving forward, Galley will be the only component that will talk with Kubernetes. We should move this to galley/pkg/source/kube/common.

I've some review comments as well. Hope this helps.

Reply:
@ozevren Thanks for your time to review this PR. What is the status for Galley will be the only component that will talk with k8s? will it be in 1.2?